presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -55,24 +62,35 @@
     private final JdbcClient delegate;
     private final boolean cacheMissing;
 
-    private final Cache<JdbcIdentity, Set<String>> schemaNamesCache;
+    private final Cache<SchemaNamesCacheKey, Set<String>> schemaNamesCache;
     private final Cache<TableNamesCacheKey, List<SchemaTableName>> tableNamesCache;
     private final Cache<TableHandleCacheKey, Optional<JdbcTableHandle>> tableHandleCache;
     private final Cache<ColumnsCacheKey, List<JdbcColumnHandle>> columnsCache;
+    private final List<PropertyMetadata> sessionPropertiesProviders;
 
     @Inject
-    public CachingJdbcClient(@StatsCollecting JdbcClient delegate, BaseJdbcConfig config)
+    public CachingJdbcClient(@StatsCollecting JdbcClient delegate, BaseJdbcConfig config, Set<SessionPropertiesProvider> sessionPropertiesProviders)
     {
-        this(delegate, config.getMetadataCacheTtl(), config.isCacheMissing());
+        this(delegate, sessionPropertiesProviders, config.getMetadataCacheTtl(), config.isCacheMissing());
     }
 
-    public CachingJdbcClient(JdbcClient delegate, Duration metadataCachingTtl, boolean cacheMissing)
+    public CachingJdbcClient(JdbcClient delegate, Set<SessionPropertiesProvider> sessionPropertiesProviders, Duration metadataCachingTtl, boolean cacheMissing)
     {
         this.delegate = requireNonNull(delegate, "delegate is null");
+        this.sessionPropertiesProviders = requireNonNull(sessionPropertiesProviders, "sessionPropertiesProviders is null").stream()
+            .flatMap(provider -> provider.getSessionProperties().stream())
+            .collect(toImmutableList());
+
         this.cacheMissing = cacheMissing;
 
         CacheBuilder<Object, Object> cacheBuilder = CacheBuilder.newBuilder()
                 .expireAfterWrite(metadataCachingTtl.toMillis(), TimeUnit.MILLISECONDS);
+
+        if (metadataCachingTtl.equals(succinctNanos(0L))) {
```

Comment:
this should be `metadataCachingTtl.toMillis() == 0`, since cache is configured with 1 ms precision

extract common variable to emphasize this must be the same thing

Reply:
I see this marked resolved, but i do not see this resolved

Reply:
I haven't pushed yet. Sorry